lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -346,9 +348,34 @@ def _on_start(self, event: StartEvent):
         # configure clients auth
         self.opensearch_config.set_client_auth()
 
+        deployment_desc = self.opensearch_peer_cm.deployment_desc()
+        # only start the main orchestrator if a data node is available
+        # this allows for "cluster-manager-only" nodes in large deployments
+        # workflow documentation:
+        # no "data" role in deployment desc -> start gets deferred
+        # when "data" node joins -> start cluster-manager via _on_peer_cluster_relation_changed
+        # cluster-manager notifies "data" node via refresh of peer cluster relation data
+        # "data" node starts and initializes security index
+        if (
+            deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            and not deployment_desc.start == StartMode.WITH_GENERATED_ROLES
+            and "data" not in deployment_desc.config.roles
+        ):
+            self.status.set(BlockedStatus(PClusterNoDataNode))
+            event.defer()
+            return
+
         # request the start of OpenSearch
         self.status.set(WaitingStatus(RequestUnitServiceOps.format("start")))
-        self._start_opensearch_event.emit()
+
+        # if this is the first data node to join, start without getting the lock
+        ignore_lock = (
+            "data" in deployment_desc.config.roles
+            and self.unit.is_leader()
+            and deployment_desc.typ == DeploymentType.OTHER
+            and not self.peers_data.get(Scope.APP, "security_index_initialised", False)
```

Comment:
```suggestion
            and (
                not self.peers_data.get(Scope.APP, "security_index_initialised", False)
                or (
                    self.peers_data.get(Scope.UNIT, "STARTED")
                    and not self.opensearch.is_service_started())
                )
            )
```

Reply:
Offline conversation: In order not to mix up PR's with different scopes and goals, this topic will be postponed and addressed in a separate task. Once both this PR and #407 are merged, this will be followed-up.

It will be tracked with https://warthogs.atlassian.net/browse/DPE-5460